From b06680d7b587d77884807a027c04692ba01a0991 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Thu, 29 Nov 2018 21:01:39 -0700
Subject: [PATCH 4/6] klte-common: Manually restorecon legacy telephony
 provider symlinks

* These need a manual restorecon after create else they do not pick
  up the radio_data_file label.
* avc: denied { read } for name="databases" dev="mmcblk0p26" ino=33320
  scontext=u:r:rild:s0 tcontext=u:object_r:system_data_file:s0
  tclass=lnk_file permissive=0

Change-Id: I3286f47521467f6a1d65befec8a5c6dd7ff00183
---
 rootdir/etc/init.qcom.rc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index 67fc3a2..dc947f5 100644
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -1003,6 +1003,8 @@ on property:sys.boot_completed=1
     # Symlink directories to access telephony.db and preferred-apn.xml required by libsec-ril
     symlink /data/user_de/0/com.android.providers.telephony/databases /data/data/com.android.providers.telephony/databases
     symlink /data/user_de/0/com.android.providers.telephony/shared_prefs /data/data/com.android.providers.telephony/shared_prefs
+    restorecon /data/data/com.android.providers.telephony/databases
+    restorecon /data/data/com.android.providers.telephony/shared_prefs
 
 service charger /charger
     class charger
-- 
2.17.1

